endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
-		payloadGetter: noop,
-		messageGetter: nilReturner,
-		expectedCode:  http.StatusOK,
-		aliased:       true,
+func jsonParseTestFile(t *testing.T, fileData []byte, testFile string) *testCaseData {
+	t.Helper()
+
+	parsedTestData := &testCaseData{}
+	var err error
+
+	// Get testCaseData values
+	parsedTestData.BidRequest, _, _, err = jsonparser.Get(fileData, "mockBidRequest")
+	assert.NoError(t, err, "Error jsonparsing root.mockBidRequest from file %s. Desc: %v.", testFile, err)
+
+	parsedReturnCode, err := jsonparser.GetInt(fileData, "expectedReturnCode")
+	assert.NoError(t, err, "Error jsonparsing root.code from file %s. Desc: %v.", testFile, err)
+
+	parsedTestData.ExpectedReturnCode = int(parsedReturnCode)
+
+	if parsedTestData.ExpectedReturnCode != 200 {
+		// Get expected error, fail if parsing error
+		parsedTestData.ExpectedErrorMessage, err = jsonparser.GetString(fileData, "expectedErrorMessage")
+		assert.NoError(t, err, "Error jsonparsing root.expectedErrorMessage from file %s. Desc: %v.", testFile, err)
+	} else {
+		// Get expected response, fail if parsing error
+		parsedTestData.ExpectedBidResponse, _, _, err = jsonparser.Get(fileData, "expectedBidResponse")
+		assert.NoError(t, err, "Error jsonparsing root.expectedBidResponse from file %s. Desc: %v.", testFile, err)
 	}
-	tests.assert(t)
-}
 
-// TestDisabledBidders makes sure we don't break when encountering a disabled bidder
-func TestDisabledBidders(t *testing.T) {
-	badTests := &getResponseFromDirectory{
-		dir:             "sample-requests/disabled/bad",
-		payloadGetter:   getRequestPayload,
-		messageGetter:   getMessage,
-		expectedCode:    http.StatusBadRequest,
-		aliased:         false,
-		disabledBidders: []string{"appnexus", "rubicon"},
-		adaptersConfig: map[string]config.Adapter{
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+	// Get testCaseConfig values
+	parsedTestData.Config = &testConfigValues{}
+	accReq, err := jsonparser.GetBoolean(fileData, "mockConfig", "accountRequired")
+	if err == nil {
+		parsedTestData.Config.AccountReq = accReq
 	}
-	goodTests := &getResponseFromDirectory{
-		dir:             "sample-requests/disabled/good",
-		payloadGetter:   noop,
-		messageGetter:   nilReturner,
-		expectedCode:    http.StatusOK,
-		aliased:         false,
-		disabledBidders: []string{"appnexus", "rubicon"},
-		adaptersConfig: map[string]config.Adapter{
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+
+	aliases, err := jsonparser.GetString(fileData, "mockConfig", "aliases")
+	if err == nil {
+		parsedTestData.Config.AliasJSON = aliases
 	}
-	badTests.assert(t)
-	goodTests.assert(t)
+
+	parsedTestData.Config.BlacklistedAccountArr = parseStringArray(fileData, "blacklistedAccts")
+	parsedTestData.Config.BlacklistedAppArr = parseStringArray(fileData, "blacklistedApps")
+	parsedTestData.Config.AdapterList = parseStringArray(fileData, "disabledAdapters")
+
+	return parsedTestData
```

Comment:
One call to jsonparser is faster than the built in library, but the benefit becomes a bit more dubious over multiple calls. Why not just use the built in library?

Reply:
We can't because some of the files have invalid json. As a compromise, we agreed to explore using json.unmarshal only for the mockConfig.

Reply:
Done. The code indeed looks cleaner by unmarshaling the `config`